### DISCUSSION TITLE
Is it possible to use Data from MongoDB for translation

### DISCUSSION THREAD

**Discussion by premnkl** [i18next/i18next#1835] [Q&A]
Dear Team,

Is it possible to use i18next by storing translation content in a react state variable ?

since our translation content is expected to be updated on runtime by user. we looking for an option to either update translation.json file dynamically or to use the value from MongoDB directly for translation

Regards,
Prem

**Comment by jamuhl** [comment]
you can use any kind of API as backend as long you provide this backend structure: https://www.i18next.com/misc/creating-own-plugins#backend

for samples have a look at: https://www.i18next.com/overview/plugins-and-utils#backends

**Reply by premnkl** [reply]
thanks for the quick response @jamuhl 

let me try the options available

**Reply by premnkl** [reply]
@jamuhl 
while I am going through the possible options, I found **addResourceBundle**

can I use this option to fetch values from DB and update the translation file whenever user adds a new value on the runtime ?

Regards,
Prem

**Reply by jamuhl** [reply]
you can use any function on the store

**Reply by premnkl** [reply]
Hi @jamuhl ,

I have a question. 

I tried using **addResourceBundle** for adding options on the runtime. But when I refresh the page, i18next keeps only the value used during initialization.

can we use **addResourceBundle** to update the JSON file in public folder ?
or
how can I keep the runtime values at any stage ?


Regards,
Prem

**Reply by jamuhl** [reply]
it's runtime only...how you persist that is up to you...eg...indexdb, localstorage...whatever
